test/rules/assertions/checkParamNames.js

```diff
@@ -1183,6 +1183,89 @@ export default {
         parser: typescriptEslintParser
       },
     },
+    {
+      code: `
+        /**
+         * @param foo
+         * @param foo.bar
+         */
+        function quux (bar, foo) {
+        }
+      `,
+      options: [
+        {
+          disableMissingParamChecks: false,
+        },
+      ],
+      errors: [
+        {
+          line: 3,
+          message: 'Expected @param names to be "bar, foo". Got "foo".',
+        },
+      ],
+    },
+    {
+      code: `
+        /**
+         * @param foo
+         */
+        function quux (bar, baz) {
+        }
+      `,
+      options: [
+        {
+          disableMissingParamChecks: true,
+        },
+      ],
+      errors: [
+        {
+          line: 3,
+          message: 'Expected @param names to be "bar, baz". Got "foo".',
+        },
+      ],
+    },
+    {
+      code: `
+        /**
+         * @param bar
+         * @param foo
+         */
+        function quux (foo, bar) {
+        }
+      `,
+      options: [
+        {
+          disableMissingParamChecks: true,
+        },
+      ],
+      errors: [
+        {
+          line: 3,
+          message: 'Expected @param names to be "foo, bar". Got "bar, foo".',
+        },
+      ],
+    },
+    {
+      code: `
+        /**
+         * @param foo
+         * @param bar
+         */
+        function quux (foo) {
+        }
+      `,
+      options: [
+        {
+          disableMissingParamChecks: true,
+        },
```

Comment:
If the option lets one skip arguments at the beginning, why not at the end?

Reply:
Oh, nevermind, I got confused.

Reply:
**Done**

I've found the test data without explaining the high-level behaviours confusing too. I stopped myself from adding comments like this:

```js
// When the disableMissingParamChecks is true, all `@param`s can be omitted.

// ..., the first `@param` can be omitted.

// ..., the last `@param` can be omitted.
```